advanced_alchemy/filters.py

```diff
@@ -202,14 +202,14 @@ def append_to_lambda_statement(
                 lambda s: s.where(any_(values) == field),  # type: ignore[arg-type]
                 track_bound_values=True,
                 track_closure_variables=False,
-                track_on=[id(values), id(model)],
+                track_on=[self.values, self.field_name, model],
             )
         values = self.values
         return statement.add_criteria(
             lambda s: s.where(field.in_(values)),
             track_bound_values=True,
             track_closure_variables=False,
-            track_on=[id(values), id(model)],
+            track_on=[self.values, self.field_name, model],
```

Comment:
If we track on `self.values`, I think this would cause a bug if any item in there isn't hashable 👀